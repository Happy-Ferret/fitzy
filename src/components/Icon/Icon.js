import PropTypes from 'prop-types';
import React from 'react';

import FitzyIcon from './Icon.styled';
import icons from './icons';

function Icon({ icon, use, size, className }) {
  return (
    <FitzyIcon
      use={use}
      size={size}
      viewBox={icons[icon].viewBox}
      className={className}
    >
      <path d={icons[icon].path} />
    </FitzyIcon>
  );
}

Icon.propTypes = {
  /**
   * Name of the icon to use
   */
  icon: PropTypes.string.isRequired,
  /**
   * Style for the icon to use
   */
  use: PropTypes.oneOf([
    'current',
    'Primary',
    'Secondary',
    'Warning',
    'Error',
    'Info',
    'Light'
  ]),
  /**
   * Size of the icon
   */
  size: PropTypes.string
};

Icon.defaultProps = {
  use: 'current',
  size: '24'
};

export default Icon;
