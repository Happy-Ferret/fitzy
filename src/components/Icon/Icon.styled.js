import styled from 'styled-components';

const FitzyIcon = styled.svg`
  display: inline-block;
  fill: ${p => (p.use === 'current' ? 'currentColor' : `${p.theme[p.use].bg}`)};
  height: ${p => `${p.size}px`};
  width: ${p => `${p.size}px`};
  vertical-align: middle;
`;

export default FitzyIcon;
