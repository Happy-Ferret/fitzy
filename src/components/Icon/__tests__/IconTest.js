import React from 'react';
import {
  mountWithTheme,
  renderWithTheme
} from '@sfitzpatrick/enzyme-context-helpers';

import Icon from '../Icon';
import icons from '../../Icon/icons';

describe('Icons', () => {
  let wrapper, component;

  const icon = 'close';

  const defaultProps = {
    icon,
    size: '16'
  };

  beforeEach(() => {
    component = <Icon {...defaultProps} />;
    wrapper = mountWithTheme(component);
  });

  test('should match snapshot', () => {
    const snap = renderWithTheme(component).toJSON();
    expect(snap).toMatchSnapshot();
    expect(snap).toHaveStyleRule('fill', 'currentColor');
    expect(snap).toHaveStyleRule('height', '16px');
    expect(snap).toHaveStyleRule('width', '16px');
  });

  test('should render Icon', () => {
    expect(wrapper.is(Icon)).toBeTruthy();
  });

  test('should render correct SVG for requested icon', () => {
    expect(wrapper.find('svg')).toHaveLength(1);
    expect(wrapper.find('path').props().d).toEqual(icons[icon].path);
  });
});
