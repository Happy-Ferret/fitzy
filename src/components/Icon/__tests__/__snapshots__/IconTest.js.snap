// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons should match snapshot 1`] = `
.c0 {
  display: inline-block;
  fill: currentColor;
  height: 16px;
  width: 16px;
  vertical-align: middle;
}

<svg
  className="c0"
  size="16"
  viewBox="0 0 24 24"
>
  <path
    d="M16.24 14.83a1 1 0 0 1-1.41 1.41L12 13.41l-2.83 2.83a1 1 0 0 1-1.41-1.41L10.59 12 7.76 9.17a1 1 0 0 1 1.41-1.41L12 10.59l2.83-2.83a1 1 0 0 1 1.41 1.41L13.41 12l2.83 2.83z"
  />
</svg>
`;
